kubetest/gcedind/gce_dind.go

```diff
@@ -0,0 +1,622 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file implements a Kubernetes kubetest deployer based on the scripts
+// in the Mirantis/kubeadm-dind-cluster github repo. These scripts are used
+// to create a containerized, Docker-in-Docker (DinD) based Kubernetes cluster
+// that is running in a remote GCE instance.
+
+package gcedind
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"k8s.io/test-infra/kubetest/process"
+)
+
+var (
+	// gce-dind specific test config flags.
+	gceDinDScriptsPath = flag.String("gce-dind-scripts-path", "",
```

Comment:
Let's try to avoid defining flags in packages. These are effectively globals.

Reply:
I eliminated most flags, and removed remainder to main.go